pkg/common/upgrade/managed.go

```diff
@@ -0,0 +1,273 @@
+func createUpgradeConfig(channel string, version string, h *helper.H) error {
+
+	// Delete any existing UpgradeConfig
+	h.Dynamic().Resource(schema.GroupVersionResource{
+		Group: "upgrade.managed.openshift.io", Version: "v1alpha1", Resource: "upgradeconfigs",
+	}).Namespace(muoNamespace).Delete(context.TODO(), upgradeConfigName, metav1.DeleteOptions{})
+
+	// Create a new UpgradeConfig and add it to the cluster
+	uc := upgradev1alpha1.UpgradeConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "UpgradeConfig",
+			APIVersion: "upgrade.managed.openshift.io/v1alpha1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      upgradeConfigName,
+			Namespace: muoNamespace,
+		},
+		Spec: upgradev1alpha1.UpgradeConfigSpec{
+			Desired: upgradev1alpha1.Update{
+				Version: version,
+				Channel: channel,
+				Force:   false,
+			},
+			UpgradeAt:            time.Now().UTC().Format(time.RFC3339),
+			PDBForceDrainTimeout: muoPdbDrainTimeout,
+			Proceed:              true,
+			Type:                 muoUpgradeType,
+		},
+	}
+
+	obj, err := runtime.DefaultUnstructuredConverter.ToUnstructured(uc.DeepCopy())
+	if err != nil {
+		return fmt.Errorf("can't convert UpgradeConfig to unstructured resource: %v", err)
+	}
+
+	uobj := unstructured.Unstructured{obj}
+	_, err = h.Dynamic().Resource(schema.GroupVersionResource{
+		Group:    "upgrade.managed.openshift.io",
+		Version:  "v1alpha1",
+		Resource: "upgradeconfigs",
+	}).Namespace(muoNamespace).Create(context.TODO(), &uobj, metav1.CreateOptions{})
+
+	if err != nil {
+		return fmt.Errorf("can't create UpgradeConfig resource: %v", err)
+	}
+
+	return nil
+}
+
+// creates workloads that are protected by Pod Disruption Budgets (PDBs)
+// which the managed-upgrade-operator is expected to be able to handle
+func createPodDisruptionBudgetWorkloads(h *helper.H) error {
+
+	if _, ok := h.GetWorkload(pdbWorkloadName); ok {
+		return nil
+	}
+
+	// Create all K8s objects that are within the testDir
+	log.Printf("Applying Pod Disruption Budget workload: %s\n", pdbWorkloadDir)
+	_, err := helper.ApplyYamlInFolder(pdbWorkloadDir, h.CurrentProject(), h.Kube())
+	if err != nil {
+		return fmt.Errorf("can't create upgrade PDB workload: %v", err)
+	}
+
+	// Give the cluster a second to churn before checking
+	time.Sleep(3 * time.Second)
+
+	// Wait for all pods to come up healthy
+	err = wait.PollImmediate(15*time.Second, 10*time.Minute, func() (bool, error) {
+		if check, err := healthchecks.CheckPodHealth(h.Kube().CoreV1(), nil); !check || err != nil {
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return fmt.Errorf("PDB workload not running correctly: %v", err)
+	}
+
+	// If success, add the workload to the list of installed workloads
+	h.AddWorkload(pdbWorkloadName, h.CurrentProject())
+
+	return nil
+}
+
+// IsManagedUpgradeDone returns with done true when a managed upgrade is complete.
+func IsManagedUpgradeDone(h *helper.H, desired *configv1.Update) (done bool, msg string, err error) {
```

Comment:
Does this need to be exported since it's all in the same package?

Reply:
Not at all - fixed in 2fbe0d2